src/App.tsx

```diff
@@ -29,67 +30,121 @@ type ReorderOptions = {
 ) {
   // To avoid the original array mutation
   const visibleGoods = [...goods];
+  const { isReversed, sortType } = options;
 
   // Sort and reverse goods if needed
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
-  return visibleGoods;
+  visibleGoods.sort((prevGood, currentGood) => {
+    const {
+      ALPHABET,
+      LENGTH,
+      NONE,
+    } = SortType;
+
+    switch (sortType) {
+      case ALPHABET:
+        return prevGood.localeCompare(currentGood);
+      case LENGTH:
+        return prevGood.length - currentGood.length;
+      case NONE:
+        return 0;
+      default: return 0;
+    }
+  });
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const {
+      LENGTH,
+      ALPHABET,
+      NONE,
+    } = SortType;
+    const visibleReset = (isReversed || sortType !== NONE);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info', {
+              'is-light': sortType !== ALPHABET,
+            })}
+            onClick={() => (this.setState({ sortType: ALPHABET }))}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-success', {
+              'is-light': sortType !== LENGTH,
+            })}
+            onClick={() => (this.setState({ sortType: LENGTH }))}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-warning', {
+              'is-light': !isReversed,
+            })}
+            onClick={() => (
+              isReversed
+                ? this.setState({ isReversed: false })
+                : this.setState({ isReversed: true })
```

Comment:
1 - move to separate method
2 - it's a boolean value, every time you click at this button this value should be changed to opposite. Don't forget about functional setState, just toggle this value, no need to use ternary operator here